### PR TITLE
Add field annotations for high-priority Syndeo schema transforms

### PR DESCRIPTION
**Pull request by andreigurau**
Add field annotations for BigQuery Storage Read, File Write, Spanner Changestreams Read, and Spanner Write schema transforms. This is necessary for tying a description to each field, so that when a schema transform config is generated, each field would have a description of what it is supposed to do

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/java/io/file-schema-transform/src/main/java/org/apache/beam/sdk/io/fileschematransform/FileWriteSchemaTransformConfiguration.java [hunk 1]
```diff
@@ -51,43 +52,41 @@ public static XmlConfiguration.Builder xmlConfigurationBuilder() {
...
+  @SchemaFieldDescription("A common prefix to use for all generated filenames.")
   public abstract String getFilenamePrefix();
 
-  /**
-   * The compression of all generated shard files. By default, appends the respective extension to
-   * the filename. See {@link org.apache.beam.sdk.io.Compression} for expected values.
-   */
+  /** See {@link org.apache.beam.sdk.io.Compression} for expected values. */
+  @SchemaFieldDescription(
+      "The compression of all generated shard files. By default, appends the respective extension to the filename.")
```
Comment (ahmedabu98) [inline sdks/java/io/file-schema-transform/src/main/java/org/apache/beam/sdk/io/fileschematransform/FileWriteSchemaTransformConfiguration.java:64]: Can you add a link to https://beam.apache.org/releases/javadoc/current/org/apache/beam/sdk/io/Compression.html for expected inputs?

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryDirectReadSchemaTransformProvider.java [hunk 2]
```diff
@@ -123,15 +124,22 @@ public static Builder builder() {
           .Builder();
     }
 
+    @SchemaFieldDescription("The SQL query to be executed to read from the BigQuery table.")
     @Nullable
     public abstract String getQuery();
 
+    @SchemaFieldDescription(
+        "The BigQuery table to read from. Format: [${PROJECT}:]${DATASET}.${TABLE}")
```
Comment (ahmedabu98) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryDirectReadSchemaTransformProvider.java:132]:
```suggestion
        "The fully-qualified name of the BigQuery table to read from. Format: [${PROJECT}:]${DATASET}.${TABLE}")
```

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryDirectReadSchemaTransformProvider.java [hunk 3]
```diff
@@ -123,15 +124,22 @@ public static Builder builder() {
...
 
+    @SchemaFieldDescription(
+        "The Google standard SQL expression to read only the rows matched from the specified filter. If no value is specified, then all rows are returned.")
     @Nullable
     public abstract String getRowRestriction();
 
+    @SchemaFieldDescription(
+        "Read only the specified fields (columns) from a BigQuery table. Fields may not be returned in the order specified. If no value is specified, then all fields are returned.")
     @Nullable
     public abstract List<String> getSelectedFields();
```
Comment (ahmedabu98) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryDirectReadSchemaTransformProvider.java:144]: How does Syndeo allow users to input a list? should that be mentioned in the description?

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryDirectReadSchemaTransformProvider.java [hunk 4]
```diff
@@ -123,15 +124,22 @@ public static Builder builder() {
...
     @Nullable
     public abstract String getQuery();
 
+    @SchemaFieldDescription(
+        "The BigQuery table to read from. Format: [${PROJECT}:]${DATASET}.${TABLE}")
     @Nullable
     public abstract String getTableSpec();
 
+    @SchemaFieldDescription(
+        "The Google standard SQL expression to read only the rows matched from the specified filter. If no value is specified, then all rows are returned.")
```
Comment (ahmedabu98) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryDirectReadSchemaTransformProvider.java:137]:
```suggestion
        "Read only rows that match this filter, which must be compatible with Google standard SQL. This is not supported when reading via query.")
```

**Review by ahmedabu98 — CHANGES_REQUESTED** [review]
Thanks for adding these! left a few suggestions

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryDirectReadSchemaTransformProvider.java (hunk 2 above)
Comment (andreigurau) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryDirectReadSchemaTransformProvider.java]: done

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryDirectReadSchemaTransformProvider.java (hunk 4 above)
Comment (andreigurau) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryDirectReadSchemaTransformProvider.java]: done

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryDirectReadSchemaTransformProvider.java (hunk 3 above)
Comment (andreigurau) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryDirectReadSchemaTransformProvider.java]: done (seems like it's using commas to separate each entry within a list)

**Review by ahmedabu98 — APPROVED** [review]
LGTM, need a committer to merge this

**Review by Abacn — APPROVED** [review]

**Comment by Abacn** [comment]
`org.apache.beam.sdk.io.gcp.datastore.RampupThrottlingFnTest.testRampupThrottler` unrelated flake

**Pull request merged**